be/src/exec/parquet_builder.cpp

```diff
@@ -0,0 +1,358 @@
+        builder.compression(parquet::Compression::UNCOMPRESSED);
+        break;
+    }
+    default:
+        builder.compression(parquet::Compression::UNCOMPRESSED);
+    }
+}
+
+void ParquetBuilder::_generate_rg_writer() {
+    if (_rg_writer == nullptr) {
+        _rg_writer = _file_writer->AppendBufferedRowGroup();
+        _cur_written_rows = 0;
+    }
+}
+
+#define DISPATCH_PARQUET_NUMERIC_WRITER(WRITER, COLUMN_TYPE, NATIVE_TYPE)                                         \
+    ParquetBuilder::_generate_rg_writer();                                                                        \
+    parquet::WRITER* col_writer = static_cast<parquet::WRITER*>(_rg_writer->column(i));                           \
+    col_writer->WriteBatch(                                                                                       \
+            num_rows, nullable ? def_level.data() : nullptr, nullptr,                                             \
+            reinterpret_cast<const NATIVE_TYPE*>(down_cast<const COLUMN_TYPE*>(data_column)->get_data().data())); \
+    _buffered_values_estimate[i] = col_writer->EstimatedBufferedValueBytes();
+
+Status ParquetBuilder::add_chunk(Chunk* chunk) {
+    if (!chunk->has_rows()) {
+        return Status::OK();
+    }
+
+    size_t num_rows = chunk->num_rows();
+    for (size_t i = 0; i < chunk->num_columns(); i++) {
+        auto& col = chunk->get_column_by_index(i);
+        bool nullable = col->is_nullable();
+        auto null_column = nullable && down_cast<NullableColumn*>(col.get())->has_null()
+                                   ? down_cast<NullableColumn*>(col.get())->null_column()
+                                   : nullptr;
+        const auto data_column = ColumnHelper::get_data_column(col.get());
+
+        std::vector<int16_t> def_level(num_rows);
+        std::fill(def_level.begin(), def_level.end(), 1);
+        if (null_column != nullptr) {
+            auto nulls = null_column->get_data();
+            for (size_t j = 0; j < num_rows; j++) {
+                def_level[j] = nulls[j] == 0;
+            }
+        }
+
+        const auto type = _output_expr_ctxs[i]->root()->type().type;
+        switch (type) {
+        case TYPE_BOOLEAN: {
+            DISPATCH_PARQUET_NUMERIC_WRITER(BoolWriter, BooleanColumn, bool)
+            break;
+        }
+        case TYPE_INT: {
+            DISPATCH_PARQUET_NUMERIC_WRITER(Int32Writer, Int32Column, int32_t)
+            break;
+        }
+        case TYPE_BIGINT: {
+            DISPATCH_PARQUET_NUMERIC_WRITER(Int64Writer, Int64Column, int64_t)
+            break;
+        }
+        case TYPE_FLOAT: {
+            DISPATCH_PARQUET_NUMERIC_WRITER(FloatWriter, FloatColumn, float)
+            break;
+        }
+        case TYPE_DOUBLE: {
+            DISPATCH_PARQUET_NUMERIC_WRITER(DoubleWriter, DoubleColumn, double)
+            break;
+        }
+        default: {
+            return Status::InvalidArgument("Unsupported type");
```

Comment:
Please support most often use types.

Reply:
we will support each type in detail later. this pr is a basic version.